golem-worker-executor-base/src/durable_host/dynamic_linking/wasm_rpc.rs

```diff
@@ -788,14 +787,25 @@ async fn grpc_call<Ctx: WorkerCtx + HostWasmRpc + ResourceStore + Send>(
                 .to_case(Case::Pascal)
         );
 
-        let service_descriptor = descriptor_pool
-            .get_service_by_name(&service_full_name)
-            .unwrap();
+        let Some(service_descriptor) = descriptor_pool
```

Comment:
You have used `wasm_rpc.rs` for your gRPC code. But this is poor code organization.

The `wasm_rpc.rs` module should ONLY contain dynamic linking code for wasm_rpc protocol, which is NOT the gRPC protocol.

You should make a new `grpc.rs`, which contains all the linking code for gRPC, and you should use that at a higher level in an appropriate way.

Reply:
sure will separate the logic into grpc.rs